torch_xla/csrc/device.h

```diff
@@ -4,44 +4,40 @@
 #include <string>
 
 #include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch/csrc/lazy/backend/backend_device.h"
 #include "torch/csrc/lazy/core/hash.h"
 
 namespace torch_xla {
 
-enum class DeviceType { CPU, GPU, TPU };
+enum class TorchXLADeviceType { CPU, GPU, TPU };
 
-struct Device {
-  Device() = default;
-  explicit Device(const std::string& device_spec);
-  Device(DeviceType hw_type, int ordinal)
-      : hw_type(hw_type), ordinal(ordinal) {}
-
-  bool operator==(const Device& other) const { return compare(other) == 0; }
+struct DeviceType : public torch::lazy::BackendDeviceType {
+  TorchXLADeviceType hw_type;
 
-  bool operator!=(const Device& other) const { return compare(other) != 0; }
-
-  bool operator<(const Device& rhs) const { return compare(rhs) < 0; }
-
-  int compare(const Device& rhs) const {
-    if (hw_type != rhs.hw_type) {
-      return hw_type < rhs.hw_type ? -1 : +1;
-    }
-    return ordinal < rhs.ordinal ? -1 : (ordinal > rhs.ordinal ? +1 : 0);
```

Comment:
I think GPU test failure is due to one of the compare function mismatch caused the different order in here vs in XRT runtime, ie "GPU:0, GPU:1" vs "GPU:1, GPU:0". @wonjoolee95 Can you spend sometime to debug this issue?

Reply:
Thanks for pointing this out! It seems like this `compare` function is exactly same in both `torch_xla` and `torch::lazy`, however the failure is still here. I'll continue to debug this.